circuitx/overlays/src/commonMain/kotlin/com/slack/circuitx/overlays/BasicAlertDialogOverlay.kt

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuitx.overlays
+
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.window.DialogProperties
+import com.slack.circuit.overlay.Overlay
+import com.slack.circuit.overlay.OverlayNavigator
+
+/** An overlay that shows an basic [AlertDialog]. */
+@ExperimentalMaterial3Api
+public class BasicAlertDialogOverlay<Model : Any, Result : Any>(
+  private val model: Model,
+  private val onDismiss: (() -> Result)? = null,
+  private val properties: DialogProperties = DialogProperties(),
+  private val content: @Composable (Model, OverlayNavigator<Result>) -> Unit,
+) : Overlay<Result> {
+  @Composable
+  override fun Content(navigator: OverlayNavigator<Result>) {
+    AlertDialog(
+      content = { content(model, navigator::finish) },
+      properties = properties,
+      onDismissRequest = {
+        // This is apparently as close as we can get to an "onDismiss" callback, which
+        // unfortunately has no animation
+        navigator.finish(onDismiss!!.invoke())
```

Comment:
With the default `DialogProperties` having the dismiss callback enabled, maybe `onDismiss` should be required?

Reply:
tbh I never know what the best thing to do is here. I guess that's at least the safest option

Reply:
[`572ab48` (#1120)](https://github.com/slackhq/circuit/pull/1120/commits/572ab487bc72f910cb88a7c954be1f66cfe5a45d)